src/joint_value_goal.cpp

```diff
@@ -0,0 +1,89 @@
+
+Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following
+disclaimer in the documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products
+derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/* Author: Karl Kruusamäe */
+/* E-mail: karl.kruusamae@ut.ee */
+
+#include <ros/ros.h>
+#include <moveit/move_group_interface/move_group_interface.h>
+#include <vector>
+
+int main(int argc, char** argv)
+{
+  // Basic ROS setup
+  // ^^^^^^^^^^^^^^^
+  ros::init(argc, argv, "joint_value_goal");
+  ros::NodeHandle node_handle;
+  ros::AsyncSpinner spinner(1);
+  spinner.start();
+
+  // MoveGroupInterface API can be found at:
+  // http://docs.ros.org/kinetic/api/moveit_ros_planning_interface/html/classmoveit_1_1planning__interface_1_1MoveGroupInterface.html
+
+  // MoveIt setup
+  // ^^^^^^^^^^^^
+  // A MoveGroupInterface instance (a client for the MoveGroup action) can be easily setup using just the name of the
+  // planning group you would like to control and plan for.
+  moveit::planning_interface::MoveGroupInterface move_group("xarm7");
+
+  // Planning to a joint value goal
+  // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  // We can plan a motion for this group by specifying target values for its joints. The MoveGroupInterface API provides
+  // several ways for setting the joint-space goal but in this example a vector containing target values for all the
+  // joints is used.
+
+  // Getting the current joint values of the robot.
+  std::vector<double> joint_values = move_group.getCurrentJointValues();
+
+  // Modifying the 4th joint value to 1.57 radians (90 degrees).
+  // Feel free to modify the joint number, target value, and the number of influenced joints.
+  joint_values[3] = 1.57;
```

Comment:
+1 for the question.

Reply:
firstly, i don't feel very strongly either way. but:

1. if i could do all the examples (incl pose goal and cartesian path) using absolute values, i probably would because i think it makes learning a tad bit easier, especially for newcomers to computer programming. i would even argue that using absolute values in pose_goal would help introducing the full capacity of geomtery_msgs/Pose and the concept of orientation early on. it's just that at the same time i hope that this package is easily adoptable for other robots. so especially in cartesian space, where guessing the reachable poses can be very tedious, relative approach might be more fool proof - simply make the incremental change small. hard-coded absolute values that result in highly illustrative pose for xarm7 would not work for, say, yaskawa sia5d and it might be very time consuming to "guess" values for reachable pose.
2. in joint-space, however, getting the absolute values right is much, much easier because your typical robot takes joint values in the range of -pi ... pi. hence, guessing workable values is fairly simple.
3. also, it might be an additional learning point to show in different demonstrations different ways of assigning values.

all in all, i could switch this demo to incremental/relative approach as well. if you guys think it would be better or more consistent that way. as i said in the beginning, i don't really feel very strongly about it.

p.s. i actually intended to do the joint value target demo differently, using joint name and target value as argument, e.g., `move_group.setJointValueTarget("joint4", 1.57);`. in my opinion that would have been extremely intuitive. however, it turned out changing only one joint value this way would result in 0 values for other joints and not keep them as is, which i found terribly unintuitive and, hence, confusing to the potential learners.

Reply:
I also have mixed feelings about it (50/50 to be exact), so in my opinion just keep it as it is.

Reply:
@Veix123 we need you final thoughts on this matter!